### PR TITLE
Fix compilation with Qt6

### PR DESCRIPTION
**Pull request by JBBgameich**
Before opening a pull-request please do:
- [x] Documentation:
  - [x] Document every new public class and function
  - [x] Add `\since QXmpp 1.X` to newly added classes and functions
  - [x] Fix any doxygen warnings from your code (see log when building with `-DBUILD_DOCUMENTATION=ON`)
- [x] When implementing or updating XEPs add it to `doc/xep.doc`
- [x] Add unit tests for everything you've changed or added 
- [x] On the top level, run `clang-format -i src/<edited-file(s)> tests/<edited-file(s)>`

### CODE REVIEW

**Comment by melvo** [comment]
Thanks for fixing this!

File: src/client/QXmppAtmManager.cpp [hunk 1]
```diff
@@ -409,20 +409,22 @@ QFuture<void> QXmppAtmManager::authenticate(const QString &encryption, const QMu
     } else {
         auto future = m_trustStorage->setTrustLevel(encryption, keyIds, QXmppTrustStorage::Authenticated);
         await(future, this, [=]() {
-            if (m_trustStorage->securityPolicy(encryption) == QXmppTrustStorage::Toakafa) {
-                auto future = distrustAutomaticallyTrustedKeys(encryption, keyIds.uniqueKeys());
-                await(future, this, [=]() {
+            await(m_trustStorage->securityPolicy(encryption), this, [=](auto securityPolicy) {
```
Comment (melvo) [inline src/client/QXmppAtmManager.cpp:412]:
```suggestion
            await(m_trustStorage->securityPolicy(encryption), this, [=](const auto securityPolicy) {
```

**Comment by melvo** [comment]
Closes #380

**Review by lnjX — APPROVED** [review]
thx

**Pull request merged**